ci/setup_grpc.sh

```diff
@@ -59,22 +77,26 @@ git clone --depth=1 -b ${install_grpc_version} https://github.com/grpc/grpc
 pushd grpc
 git submodule init
 git submodule update --depth 1
-mkdir -p "third_party/abseil-cpp/build" && pushd "third_party/abseil-cpp/build"
-set -x
 
-ABSEIL_CPP_BUILD_OPTIONS=(
-    -DCMAKE_BUILD_TYPE=Release
-    -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE
-    -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR
-)
-if [ ! -z "$build_shared_libs" ]; then
-    ABSEIL_CPP_BUILD_OPTIONS=(${ABSEIL_CPP_BUILD_OPTIONS[@]} "-DBUILD_SHARED_LIBS=$build_shared_libs")
+if [[ $build_internal_abseil_cpp -ne 0 ]]; then
+    mkdir -p "third_party/abseil-cpp/build" && pushd "third_party/abseil-cpp/build"
+    set -x
+
+    ABSEIL_CPP_BUILD_OPTIONS=(
+        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE
+        -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR
+    )
+    if [ ! -z "$build_shared_libs" ]; then
+        ABSEIL_CPP_BUILD_OPTIONS=(${ABSEIL_CPP_BUILD_OPTIONS[@]} "-DBUILD_SHARED_LIBS=$build_shared_libs")
+    fi
+    cmake  ${ABSEIL_CPP_BUILD_OPTIONS[@]} ..
+    cmake --build . -j${nproc} --target install && popd
```

Comment:
Could all the logic related to abseil be moved to install_abseil.sh instead ?

Reply:
I'm not sure, but some CI jobs use `setup_grpc.sh` but without `install_abseil.sh`. This installation existed before, I only add a option to allow gRPC find abseil-cpp from a existed installation path.
Maybe we can optimize abseil-cpp installation in another PR?